source/Cosmos.Core/Memory/HeapLarge.cs

```diff
@@ -35,8 +35,7 @@ public static void Init()
             var xPtr = (uint*)RAT.AllocPages(aType, xPages);
             if (xPtr == null)

```

Comment:
Shouldn't the size of `xPtr` to be checked along with `null` check?

Reply:
What do you mean with size of `xPtr`?

Reply:
I mean `xPtr.Length()` to be 2

Reply:
Length is not defined for a pointer?

Reply:
since there's attempt to access the array positions 0 and 1, would be part of sanity to check the length of array along with null check (which is being done already).

Reply:
This is not possible for pointers. Pointers point to raw areas of memory; bound checking is not possible.